src/main/java/dk/cngroup/lentils/controller/ClientController.java

```diff
@@ -0,0 +1,101 @@
+package dk.cngroup.lentils.controller;
+
+import dk.cngroup.lentils.entity.Cypher;
+import dk.cngroup.lentils.entity.CypherStatus;
+import dk.cngroup.lentils.service.CypherService;
+import dk.cngroup.lentils.service.HintService;
+import dk.cngroup.lentils.service.HintTakenService;
+import dk.cngroup.lentils.service.StatusService;
+import dk.cngroup.lentils.service.TeamService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+@RequestMapping("/client")
+public class ClientController {
+
+    private static final String CLIENT_VIEW_CYPHER_LIST = "client/list";
+    private static final String CLIENT_VIEW_CYPHER_DESCRIPTION = "client/detail";
+    private static final String REDIRECT_CLIENT_VIEW_CYPHER_LIST = "redirect:/client/list";
+    private static final String REDIRECT_CLIENT_VIEW_CYPHER_DESCRIPTION = "redirect:/client/detail";
+
+    private CypherService cypherService;
+    private TeamService teamService;
+    private StatusService statusService;
+    private HintService hintService;
+    private HintTakenService hintTakenService;
+
+    @Autowired
+    public ClientController(final CypherService cypherService,
+                            final TeamService teamService,
+                            final StatusService statusService,
+                            final HintService hintService,
+                            final HintTakenService hintTakenService) {
+        this.cypherService = cypherService;
+        this.teamService = teamService;
+        this.statusService = statusService;
+        this.hintService = hintService;
+        this.hintTakenService = hintTakenService;
+    }
+
+    @GetMapping(value = "/list")
+    public String listAllCyphers(final Model model) {
+        model.addAttribute("cyphers", cypherService.getAll());
+        model.addAttribute("team", teamService.getTeam(2L));
```

Comment:
Team uz by sel vycist ze session, ale to bych uddlal zvlast. Vytvoris na to prosim issue?

Reply:
vytvořeno